src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -319,7 +319,14 @@ export class EditorPart extends Part implements IEditorPart, IEditorGroupService
 		// while the UI is not yet ready. Clients have to deal with this fact and we have to make sure that the
 		// stacks model gets updated if any of the UI updating fails with an error.
 		const group = this.ensureGroup(position, !options || !options.preserveFocus);
-		const pinned = !this.tabOptions.previewEditors || (options && (options.pinned || typeof options.index === 'number')) || input.isDirty();
+
+		let pinned = !this.tabOptions.previewEditors;
```

Comment:
This allows `pinned: false` to be honored explicitly rather than it falling through as a default.